src/dimensions.jl

```diff
@@ -232,8 +232,8 @@ virtual_intersect(ctx, a::Dimensionless, b::Dimensionless) = b
 
 function virtual_intersect(ctx, a::Extent, b::Extent)
     Extent(
-        start = @f(max($(getstart(a)), $(getstart(b)))),
-        stop = @f(min($(getstop(a)), $(getstop(b))))
+        start = simplify(@f(max($(getstart(a)), $(getstart(b)))), ctx),
```

Comment:
The CI tests on the main branch used to fail because the result of `simplify` was occasionally different on x64 Ubuntu Julia 1.6.7 when the Finch expression included a Limit. 

Therefore, I attempted to explicitly launch the `simplify` pass using this code snippet to verify if it works. In any case, it appears that your main branch now passes the continuous tests after your hash fix, so I assume the underlying issue was related to the static hash handling of the Limit type. Consequently, we can remove this change.